### ISSUE TITLE
Please add the window size parameter (and other options) to SSIM

### ISSUE THREAD

**Issue by veritas9872** [tensorflow/tensorflow#26929]
**System information**
- TensorFlow version (you are using): 1.13
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
Currently, the SSIM in tf.image.ssim has many parameters fixed inside the script with no way to alter them. Most importantly, the window size parameter is fixed at 11x11. 

However, many people have found that changing the window size when using SSIM as a loss function changes the learning outputs. (See https://arxiv.org/pdf/1511.08861.pdf from the authors who first proposed the use of SSIM as a loss function)

It would be very useful if SSIM and its relatives (e.g. MS-SSIM) could have their parameters (such as window size and sigma) set as defaults but be made alterable in the user interface. This is already the case in the implementation on skimage (skimage.measure.compare_ssim). 

**Will this change the current api? How?**
Not much. If the window size and other parameters are set as defaults, instead of being fixed values, most people will never notice the difference. There would be no backward incompatible changes.

**Who will benefit with this feature?**
People who wish to use SSIM as a loss function.

**Any Other info.**
In the current implementation many variables are fixed in the script instead of being defined in the functions. 

Moreover, functions such as SSIM in the current Tensorflow image_ops implementation are heavily intertwined with many other functions in the page.

This makes it very difficult for outsiders to simply copy-paste the necessary parts out to change only the parts that they need. 

It would be great if the API changed to allow parameter tuning of key hyperparameters such as window size.

Many thanks in advance to those who look into this issue.

**Comment by Gurpreetsingh9465** [comment]
@veritas9872 there are only 2 hyper-parameter for SSIM filter and sigma [ssim](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/image_ops_impl.py#L2912-L2913) so we are supposed to add these two as default parameter?

**Comment by Gurpreetsingh9465** [comment]
@jvishnuvardhan are you working on this issue ? 😄 would love to help

**Comment by veritas9872** [comment]
Hello. Thank you for helping out. 
@Gurpreetsingh9465 It is my understanding that there are also K1 and K2 parameters. However, I don't understand their function perfectly.
Also, it would be great if MS-SSIM (Multi-scale SSIM) could also have default parameters. 
Many thanks again for helping me out.

**Comment by Gurpreetsingh9465** [comment]
@veritas9872 sir, added k1 and k2 as default parameters 👍  [changes](https://github.com/tensorflow/tensorflow/pull/27076).  and sir can you open a new issue regarding mssim so that the code didn't clash.

**Comment by veritas9872** [comment]
@Gurpreetsingh9465 Thank you very much. 
However, I find that many of the checks were not successful. 
I will open a separate issue for MS-SSIM.
Thank you for helping out.

**Comment by veritas9872** [comment]
@Gurpreetsingh9465 I opened a separate issue for MS-SSIM at  https://github.com/tensorflow/tensorflow/issues/27138 
I also suggested that some values (such as K1 and K2) might be better left as keyword arguments (as is the case in skimage) since almost no one ever changes them.

**Comment by veritas9872** [comment]
I closed the issue on MS-SSIM because I saw that everything had already been done.
@Gurpreetsingh9465 Thank you for your help on this issue!

**Comment by jvishnuvardhan** [comment]
@veritas9872 Is this resolved or still an issue? If it was resolved, could you close it. Thanks!

**Comment by veritas9872** [comment]
Thank you. I think that this issue has been placed in a branch, though it has not been merged to master yet. 
I will close this issue. Thank you everyone for helping out.
